packages/core/core-flows/src/order/utils/maybe-refresh-shipping-methods.ts

```diff
@@ -0,0 +1,170 @@
+ * The shipping method and the action is updated if the shipping option is calculated.
+ */
+export const maybeRefreshShippingMethodsWorkflow = createWorkflow(
+  maybeRefreshShippingMethodsWorkflowId,
+  function (
+    input: MaybeRefreshShippingMethodsWorkflowInput
+  ): WorkflowResponse<void> {
+    const shippingMethod = useRemoteQueryStep({
+      entry_point: "order_shipping_method",
+      fields: ["id", "shipping_option_id"],
+      variables: {
+        id: input.shipping_method_id,
+      },
+      list: false,
+    }).config({ name: "fetch-shipping-method" })
+
+    const shippingOption = useRemoteQueryStep({
+      entry_point: "shipping_option",
+      fields: [...COMMON_OPTIONS_FIELDS],
+      variables: { id: shippingMethod.shipping_option_id },
+      list: false,
+    }).config({ name: "calculated-option" })
+
+    const isCalculatedPriceShippingOption = transform(
+      shippingOption,
+      (option) => option?.price_type === ShippingOptionPriceType.CALCULATED
+    )
+
+    when(
+      { isCalculatedPriceShippingOption, shippingOption },
+      ({ isCalculatedPriceShippingOption, shippingOption }) =>
+        isCalculatedPriceShippingOption
+    ).then(() => {
+      const order = useRemoteQueryStep({
+        entry_point: "orders",
+        fields: ["id", "shipping_address", "items.*", "items.variant.*"],
+        variables: { id: input.order_id },
+        list: false,
+        throw_if_key_not_found: true,
+      }).config({ name: "order-query" })
+
+      const calculateShippingOptionsPricesData = transform(
+        {
+          shippingOption,
+          order,
+          input,
+        },
+        ({ shippingOption, order, input }) => {
+          return [
+            {
+              id: shippingOption.id as string,
+              optionData: shippingOption.data,
+              context: {
+                ...order,
+                ...input.context,
+                from_location:
+                  shippingOption.service_zone.fulfillment_set.location,
+              },
+              // data: {}, // TODO: add data
+              provider_id: shippingOption.provider_id,
+            } as CalculateShippingOptionPriceDTO,
+          ]
+        }
+      )
+
+      const prices = calculateShippingOptionsPricesStep(
+        calculateShippingOptionsPricesData
+      )
+
+      const updateData = transform(
+        {
+          shippingOption,
+          prices,
+          input,
+        },
+        ({ prices, input }) => {
+          return [
+            {
+              id: input.action_id,
+              amount: prices[0].calculated_amount,
+            },
+            {
+              id: input.shipping_method_id,
+              amount: prices[0].calculated_amount,
+              is_custom_amount: false,
+            },
+          ]
+        }
+      )
+
+      parallelize(
+        updateOrderChangeActionsStep([updateData[0]]),
+        updateOrderShippingMethodsStep([updateData[1]!])
+      )
+    })
```

Comment:
question: Not sure I understand this logic, could I get you to explain?

Reply:
So if the shipping option is calculated, we update the amount on the method and the associated order change. Not sure if that answers your question?